modules/apps/frontend-js/frontend-js-module-launcher/src/main/java/com/liferay/frontend/js/module/launcher/internal/JSModuleLauncherImpl.java

```diff
@@ -129,27 +130,55 @@ private String _getScriptBody(
 		Collection<JSModuleDependency> jsModuleDependencies,
 		String javaScriptCode) {
 
-		StringBundler javascriptSB = new StringBundler(10);
+		StringBundler javascriptSB;
 
-		javascriptSB.append("(function() {");
+		if (jsModuleDependencies.size() == 1) {
+			javascriptSB = new StringBundler(10);
 
-		for (JSModuleDependency jsModuleDependency : jsModuleDependencies) {
-			javascriptSB.append("window[");
-			javascriptSB.append("Symbol.for('__LIFERAY_WEBPACK_GET_MODULE__')");
-			javascriptSB.append("]('");
+			javascriptSB.append("(function() {window[Symbol.for('");
+			javascriptSB.append("__LIFERAY_WEBPACK_GET_MODULE__')](");
+
+			Iterator<JSModuleDependency> iterator =
+				jsModuleDependencies.iterator();
+
+			JSModuleDependency jsModuleDependency = iterator.next();
+
+			javascriptSB.append(StringPool.APOSTROPHE);
 			javascriptSB.append(jsModuleDependency.getModuleName());
-			javascriptSB.append("').then((");
+			javascriptSB.append(StringPool.APOSTROPHE);
+
+			javascriptSB.append(").then((");
+
 			javascriptSB.append(jsModuleDependency.getVariableName());
+
 			javascriptSB.append(") => {");
+			javascriptSB.append(javaScriptCode);
+			javascriptSB.append("});})();");
 		}
+		else {
+			javascriptSB = new StringBundler(
+				6 + (5 * jsModuleDependencies.size()));
 
-		javascriptSB.append(javaScriptCode);
+			javascriptSB.append("(function() {Promise.all([");
 
-		for (int i = 0; i < jsModuleDependencies.size(); i++) {
-			javascriptSB.append("});");
-		}
+			for (JSModuleDependency jsModuleDependency : jsModuleDependencies) {
```

Comment:
The obvious question: how many of these dependencies de we expect to be?

Reply:
Not too many. As far as I've seen it shouldn't be more than three (the react renderer + react component + props transformer).

However, if someone asks for 100 modules it's their fault, not  ours :-D

Reply:
Think of this method as `open()` to open files: if you open too many files, it's not `open()`'s fault :shrug:

Reply:
3 sounds like a good number.

And if and when somebody asks for 100 we can pool/batch.